packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/platform_interface/google_maps_flutter_platform.dart

```diff
@@ -0,0 +1,316 @@
+  /// platform side.
+  ///
+  /// The returned [Future] completes after listeners have been notified.
+  Future<void> updateMarkers(
+    MarkerUpdates markerUpdates, {
+    @required int mapId,
+  }) {
+    throw UnimplementedError('updateMarkers() has not been implemented.');
+  }
+
+  /// Updates polygon configuration.
+  ///
+  /// Change listeners are notified once the update has been made on the
+  /// platform side.
+  ///
+  /// The returned [Future] completes after listeners have been notified.
+  Future<void> updatePolygons(
+    PolygonUpdates polygonUpdates, {
+    @required int mapId,
+  }) {
+    throw UnimplementedError('updatePolygons() has not been implemented.');
+  }
+
+  /// Updates polyline configuration.
+  ///
+  /// Change listeners are notified once the update has been made on the
+  /// platform side.
+  ///
+  /// The returned [Future] completes after listeners have been notified.
+  Future<void> updatePolylines(
+    PolylineUpdates polylineUpdates, {
+    @required int mapId,
+  }) {
+    throw UnimplementedError('updatePolylines() has not been implemented.');
+  }
+
+  /// Updates circle configuration.
+  ///
+  /// Change listeners are notified once the update has been made on the
+  /// platform side.
+  ///
+  /// The returned [Future] completes after listeners have been notified.
+  Future<void> updateCircles(
+    CircleUpdates circleUpdates, {
+    @required int mapId,
+  }) {
+    throw UnimplementedError('updateCircles() has not been implemented.');
+  }
+
+  /// Starts an animated change of the map camera position.
+  ///
+  /// The returned [Future] completes after the change has been started on the
+  /// platform side.
+  Future<void> animateCamera(
+    CameraUpdate cameraUpdate, {
+    @required int mapId,
+  }) {
+    throw UnimplementedError('animateCamera() has not been implemented.');
+  }
+
+  /// Changes the map camera position.
+  ///
+  /// The returned [Future] completes after the change has been made on the
+  /// platform side.
+  Future<void> moveCamera(
+    CameraUpdate cameraUpdate, {
+    @required int mapId,
+  }) {
+    throw UnimplementedError('moveCamera() has not been implemented.');
+  }
+
+  /// Sets the styling of the base map.
+  ///
+  /// Set to `null` to clear any previous custom styling.
+  ///
+  /// If problems were detected with the [mapStyle], including un-parsable
+  /// styling JSON, unrecognized feature type, unrecognized element type, or
+  /// invalid styler keys: [MapStyleException] is thrown and the current
+  /// style is left unchanged.
+  ///
+  /// The style string can be generated using [map style tool](https://mapstyle.withgoogle.com/).
+  /// Also, refer [iOS](https://developers.google.com/maps/documentation/ios-sdk/style-reference)
+  /// and [Android](https://developers.google.com/maps/documentation/android-sdk/style-reference)
+  /// style reference for more information regarding the supported styles.
```

Comment:
This seems better be put in the platform specific overrides instead of here.

Reply:
I'll leave the iOS and Android on the platform specific overrides.